src/sentry/rules/conditions/level.py

```diff
@@ -58,7 +59,12 @@ def passes(self, event, state, **kwargs):
             return False
 
         desired_level = int(desired_level)
-        level = int(event.level)
+        # Try to fetch the event level from the tags first since event.level
+        # is event.group.level which may not always be the most up to date
+        if event.get_tag('level') and LOG_LEVEL_REVERSE_MAP.get(event.get_tag('level')):
+            level = LOG_LEVEL_REVERSE_MAP.get(event.get_tag('level'))
+        else:
+            level = int(event.level)
```

Comment:
Do we even need a fallback here?  Won't all events have the level tag after being normalized in the EventManager?